src/include/function/vector_operations.h

```diff
@@ -12,6 +12,9 @@ namespace function {
 
 struct VectorOperationDefinition;
 
+using scalar_compile_func =
+    std::function<void(FunctionBindData*, const std::vector<std::shared_ptr<common::ValueVector>>&,
+        std::shared_ptr<common::ValueVector>&)>;
```

Comment:
i think we never use a reference to a shared_ptr, either change it to a raw ptr or remove reference

Reply:
This has to be a reference, because we are changing the shared_ptr inside compileFunc